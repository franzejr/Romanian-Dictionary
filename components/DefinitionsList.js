import React from 'react';
import api  from '../API';
import Separator from './Separator';

import {
  View,
  Text,
  ScrollView,
  StyleSheet,
  TouchableHighlight,
} from 'react-native';

var styles = StyleSheet.create({
  rowContainer: {
    flexDirection: 'column',
    flex: 1,
    padding: 10
  }
});

const DefinitionsList=({definitions})=>{

  var list = definitions.map((item, index) => {
     return (
       <View key={index}>
         <View style={styles.rowContainer}>
           <Text>{definitions[index].name}</Text>
         </View>
         <Separator />
       </View>
     )
   });

  return(
    <View style={styles.mainContainer}>
      {list}
    </View>
    )
};

export default DefinitionsList;
