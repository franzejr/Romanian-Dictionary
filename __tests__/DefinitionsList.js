import 'react-native';
import React from 'react';
import DefinitionsList from '../components/DefinitionsList';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <DefinitionsList definitions={['a', 'b']} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
