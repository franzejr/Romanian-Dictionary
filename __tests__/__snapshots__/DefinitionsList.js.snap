exports[`test renders correctly 1`] = `
<View
  style={undefined}>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "padding": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail" />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#E4E4E4",
          "flex": 1,
          "height": 1,
          "marginLeft": 15,
        }
      } />
  </View>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "padding": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail" />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#E4E4E4",
          "flex": 1,
          "height": 1,
          "marginLeft": 15,
        }
      } />
  </View>
</View>
`;
